src/js/src/mixins/ImageSearchUtils.js

```diff
@@ -0,0 +1,37 @@
+import HistoryRoutingUtils from './HistoryRoutingUtils'
+import { mapState, mapActions } from 'vuex'
+
+export default {
+  mixins: [HistoryRoutingUtils],
+  computed: {
+    ...mapState({
+      searchAppliedFacets: state => state.Search.searchAppliedFacets,
+      solrSettings: state => state.Search.solrSettings,
+    }),
+  },
+  methods: {
+    ...mapActions('Search', {
+      updateSearchAppliedFacets:'updateSearchAppliedFacets',
+      updateSolrSettingImgSearch:'updateSolrSettingImgSearch',
+    }),
+    $_startPageSearch(searchItem) {
+      return '/?q=' + 'links_images:"' + searchItem + '"' + '&offset=' + this.solrSettings.offset + '&grouping=' + this.solrSettings.grouping + '&imgSearch=false&urlSearch=false'
+    },
+    $_startImageSearch(searchItem) {
+      return '/?q=' + 'hash:"' + searchItem + '"' + '&offset=' + this.solrSettings.offset + '&grouping=' + this.solrSettings.grouping + '&imgSearch=false&urlSearch=false'
+    },
+    $_addHistory(field, searchItem) {
```

Comment:
Only a suggestion:  This mixin is solely related to image search so maybe if you throw an _image_ into the naming it would be less generic, easier to see the relationship to image search and less probability of naming collision.

Reply:
Very good idea, done